### PR TITLE
cargo: update to fail 0.3

### PR DESCRIPTION
**Pull request by lucab**
This updates to latest fail 0.3. It also aligns feature naming to use `failpoints` everywhere, like the
underlying library.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ members = ["proto"]
 [features]
 default = []
 # Enable failpoints
-failpoint = ["fail"]
+failpoints = ["fail", "fail/failpoints"]
```
Comment (BusyJay) [inline Cargo.toml]: I think `["fail/failpoints"]` is enough.

Comment (lucab) [reply Cargo.toml]: Indeed, I tried and that works. I just learned something new here, thanks for the suggestion!

**Comment by lucab** [comment]
Amended with @BusyJay suggestion, PTAL.

**Pull request merged**